### PR TITLE
nibble: use trait to eliminate dep for crypto

### PR DESCRIPTION
**Pull request by bmwill**

### CODE REVIEW

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  R diem-nibble 0.1.0 (workspace, path 'common/nibble')
    * (old features: default)
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  R diem-nibble 0.1.0 (workspace, path 'common/nibble')
    * (old features: default)
```

File: storage/jellyfish-merkle/src/lib.rs [hunk 1]
```diff
@@ -655,3 +656,78 @@ where
             .map(|root_node| root_node.hash()))
     }
 }
+
+trait NibbleExt {
```
Comment (kchalkias) [inline storage/jellyfish-merkle/src/lib.rs:660]: So NibbleExt is only used in Merkle trees?

Comment (bmwill) [reply storage/jellyfish-merkle/src/lib.rs:660]: Yeah these nibble functions on the `HashValue` type are only ever used in this module (and I think they were added explicitly for the merkle trees)

Comment (mimoo) [reply storage/jellyfish-merkle/src/lib.rs:660]: I applaude the cleaning

**Comment by bmwill** [comment]
@kchalkias or @mimoo any concerns about this PR?

**Review by kchalkias — APPROVED** [review]
Sweet!

**Pull request merged**